File: 路径总和.cpp
class Solution {
public:
    bool hasPathSum(TreeNode* root, int targetSum) {
        if (root == NULL) {
            return false;
        }
        targetSum -= root->val;
        if (root->left == root->right) {
            return targetSum == 0;
        }
        return hasPathSum(root->left, targetSum) or hasPathSum(root->right, targetSum);
    }
};

File: 路径总和.py
class Solution:
    def hasPathSum(self, root: Optional[TreeNode], targetSum: int) -> bool:
        if root is None:
            return False
        targetSum -= root.val
        if root.left == root.right:
            return targetSum == 0
        return self.hasPathSum(root.left, targetSum) or self.hasPathSum(root.right, targetSum)
